apps/io-profile-async/src/functions/__tests__/expired-sessions-discoverer.test.ts

```diff
@@ -0,0 +1,550 @@
+import { ExpiredUserSessionsQueueRepository } from "../../repositories/expired-user-sessions-queue";
+import { SessionNotificationsRepository } from "../../repositories/session-notifications";
+import * as appinsights from "../../utils/appinsights";
+import { TransientError } from "../../utils/errors";
+import {
+  ExpiredSessionsDiscovererFunction,
+  ItemToProcess,
+  processChunk,
+  processItem,
+  retrieveFromDbInChunks
+} from "../expired-sessions-discoverer";
+import { createInterval, Interval } from "../../types/interval";
+
+const aSession = ({
+  id: "AAAAAA89S20I111X",
+  expiredAt: Date.now(),
+  notificationEvents: {
+    EXPIRED_SESSION: false
+  },
+  _etag: "etag",
+  _rid: "rid",
+  _self: "self",
+  _ts: 0,
+  kind: "IRetrievedSessionNotifications"
+} as unknown) as RetrievedSessionNotifications;
+
+const item = {
+  itemTimeoutInSeconds: 5,
+  queuePayload: {
+    fiscalCode: aSession.id,
+    expiredAt: new Date(aSession.expiredAt)
+  },
+  retrievedDbItem: aSession
+} as ItemToProcess;
+
+const expiredSessionsDiscovererConfMock = {
+  EXPIRED_SESSION_ADVISOR_QUEUE: "aQueueName",
+  EXPIRED_SESSION_SCANNER_CHUNK_SIZE: 1,
+  EXPIRED_SESSION_SCANNER_TIMEOUT_MULTIPLIER: 1,
+  SESSION_NOTIFICATIONS_CONTAINER_NAME: "aContainerName"
+} as ExpiredSessionDiscovererConfig;
+
+const sendExpiredUserSessionMock = vi.fn(
+  () => TE.of(void 0) as TE.TaskEither<Error, void>
+);
+
+const updateExpiredSessionNotificationFlagMock = vi.fn(
+  () => TE.of(void 0) as TE.TaskEither<CosmosErrors, void>
+);
+
+const findByExpiredAtAsyncIterableMock = vi.fn(
+  () =>
+    async function*() {
+      yield [E.of(aSession)];
+    }
+);
+
+const mockExpiredUserSessionsQueueRepository = ({
+  sendExpiredUserSession: vi.fn(() => sendExpiredUserSessionMock)
+} as unknown) as ExpiredUserSessionsQueueRepository;
+
+const mockSessionNotificationsRepository = ({
+  updateExpiredSessionNotificationFlag: vi.fn(
+    () => updateExpiredSessionNotificationFlagMock
+  ),
+  findByExpiredAtAsyncIterable: findByExpiredAtAsyncIterableMock
+} as unknown) as SessionNotificationsRepository;
+
+const mockSessionsNotificationModel = ({
+  buildAsyncIterable: vi.fn(),
+  patch: vi.fn()
+} as unknown) as SessionNotificationsModel;
+
+const mockExpiredUserSessionsQueueClient = ({
+  sendMessage: vi.fn()
+} as unknown) as QueueClient;
+
+const baseDeps = {
+  expiredSessionsDiscovererConf: expiredSessionsDiscovererConfMock,
+  ExpiredUserSessionsQueueRepo: mockExpiredUserSessionsQueueRepository,
+  SessionNotificationsRepo: mockSessionNotificationsRepository,
+  sessionNotificationsModel: mockSessionsNotificationModel,
+  expiredUserSessionsQueueClient: mockExpiredUserSessionsQueueClient
+};
+
+const trackEventMock = vi.spyOn(appinsights, "trackEvent");
+
+// eslint-disable-next-line max-lines-per-function
+describe("Expired Sessions Discoverer TimerTrigger Tests", () => {
+  afterEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe("processItem", () => {
```

Comment:
One last test is missing: “It should fail with a transient error when sending to the queue fails and the item has been resumed for processing.”